forte/base_classes/state_info.cc

```diff
@@ -69,9 +69,16 @@ const std::vector<size_t>& StateInfo::gas_min() const { return gas_min_; }
 const std::vector<size_t>& StateInfo::gas_max() const { return gas_max_; }
 
 bool StateInfo::operator<(const StateInfo& rhs) const {
-    return std::tie(na_, nb_, multiplicity_, twice_ms_, irrep_, gas_min_, gas_max_) <
-           std::tie(rhs.na_, rhs.nb_, rhs.multiplicity_, rhs.twice_ms_, rhs.irrep_, rhs.gas_min_,
-                    rhs.gas_max_);
+    // Make sure the roots are in correct energy order
+    if ((gas_min_ == rhs.gas_min_) && (gas_max_ == rhs.gas_max_)) {
+        return std::tie(na_, nb_, multiplicity_, twice_ms_, irrep_, gas_min_, gas_max_) <
+               std::tie(rhs.na_, rhs.nb_, rhs.multiplicity_, rhs.twice_ms_, rhs.irrep_,
+                        rhs.gas_min_, rhs.gas_max_);
+    } else if (gas_max_ == rhs.gas_max_) {
+        return gas_min_ > rhs.gas_min_;
```

Comment:
Seems to me this should be
```suggestion
        return gas_min_ < rhs.gas_min_;
```
?

Reply:
Thanks for the clarification. Might worth writing some comments here because this is not we used to have (the tuple comparison actually does `<`).